pkg/dss/scd/cockroach/operations.go

```diff
@@ -233,8 +233,16 @@ func (s *Store) DeleteOperation(ctx context.Context, id scdmodels.ID, owner dssm
 				id = $1
 			AND
 				owner = $2
-			RETURNING
-				*
+		`
+		deleteImplicitSubscriptionQuery = `
+			DELETE FROM
+				scd_subscriptions
+			WHERE
+				id = $1
+			AND
+				owner = $2
+			AND
+				implicit = true
```

Comment:
We also need to check that the only dependent Operation is the one we're deleting.  A client could create a new Operation which triggers an implicit Subscription creation, then create another new Operation pointing to the implicit Subscription, then delete the first Operation.  In that case, the Subscription would still have a dependent Operation so it shouldn't be deleted.

Reply:
Ah, nice catch.

Reply:
Fixed.